### DISCUSSION TITLE
Need help for build / deploy

### DISCUSSION THREAD

**Discussion by seiwaldm** [antfu/vitesse#445] [Q&A] [answered]
Hi Community!

I spent a good amount of time during the last two weeks on creating this project: https://github.com/seiwaldm/heroclash-vitesse 

It's a bit of a Frankenstein and I used plain JS, but it runs nicely in dev mode. However, I fail to build the project and serve it locally from dist, and a Netlify deploy also fails. I have no prior experience in distributing such a complex project, so it might be an obvious and easy to fix mistake. But I'm unable to see it. 

Maybe someone can throw a look at it and help me out. The game itself is quite fun. It's a Vue Port and Extension of this previous Vanilla JS exercise: https://seiwaldm.github.io/heroclash/ There are quite a few ideas on how to extend this project together with students in a classroom setting. The main goal is to practise Vue basics. This can be achieved in dev mode only... but it's very frustrating and disencouraging for the students, if we are not able to actually deploy it. So help would be very much appreciated. Also pointing me to another platform, where it is more appropriate to ask for help like this - thanks for your understanding and effort!

**Comment by spokospace** [comment]
What's wrong after put pnpm build in terminal?

Can u use current netlify.toml file: https://github.com/antfu/vitesse/blob/main/netlify.toml and check logs on netlify?

**Comment by seiwaldm** [comment]
Thank you very much for the quick response @spokospace!

When I try to deploy to Netlify the deploy log tells me the following: 

> 8:49:53 PM: Found pnpm version (7.8.0) that doesn't match expected (7.13.4)
Preparing pnpm@7.13.4 for immediate activation...
8:49:53 PM: No pnpm workspaces detected
8:49:53 PM: Started restoring cached node modules
8:49:53 PM: Finished restoring cached node modules
8:49:53 PM: Installing NPM modules using PNPM version 7.8.0
8:49:54 PM: Lockfile is up-to-date, resolution step is skipped
8:49:54 PM:  ERR_PNPM_OUTDATED_LOCKFILE  Cannot install with "frozen-lockfile" because pnpm-lock.yaml is not up-to-date with package.json
8:49:54 PM: Note that in CI environments this setting is true by default. If you still need to run install in such cases, use "pnpm install --no-frozen-lockfile"
8:49:54 PM: Creating deploy upload records
8:49:54 PM: Error during PNPM install
8:49:54 PM: Build was terminated: Build script returned non-zero exit code: 1
8:49:54 PM: Failing build: Failed to build site
8:49:54 PM: Failed during stage 'building site': Build script returned non-zero exit code: 1 (https://ntl.fyi/exit-code-1)
8:49:54 PM: Finished processing build request in 6.661147281s

I guess the problem might be that I installed two packages via npm (pocketbase and pinia-plugin-persistedstate)?

---

When I try to build locally I get an error concerning the pinia-plugin-persistedstate package:

> ReferenceError: localStorage is not defined
    at file:///C:/Users/marku/Desktop/coding-beispiele-schule/heroclash-vit
esse/node_modules/.pnpm/pinia-plugin-persistedstate@2.3.0_pinia@2.0.22/node
_modules/pinia-plugin-persistedstate/dist/chunk-54L43XJA.mjs:135:19

However, the build process continues and this error should not be a problem when I try to access index.html, as I don't use localStorage on that page...(?) However, I only get the 404 layout when i try to open index.html or any other route.

Another guess might be that I didn't follow step 3 of the recommendation by antfu to adapt the template: "Change the hostname in vite.config.ts". I just didn't understand what I have to change exactly...

btw: I also tried to build with `vite build` (without ssg)... same results - 404 on every page.

thx again for taking your time to think about my problems!

**Reply by bitsnaps** [reply]
> Another guess might be that I didn't follow step 3 of the recommendation by antfu to adapt the template: "Change the hostname in vite.config.ts". I just didn't understand what I have to change exactly...

@antfu  for this, I think we need to remove it from the README, or change it to something else, because there is no default `hostname` in `vite.config.ts` file by default, right?

**Comment by seiwaldm** [comment, marked as answer]
Thank you @spokospace! I found at least the solution to deploy to netlify. just had to add pocketbase and the plugin for pinia with `pnpm add` to update pnpm-lock.yaml.... 

will try later on to deploy it to my own server. but for the moment i'm satisfied :)
